### PR TITLE
feat(reports): enable optional reports generator deployment

### LINKED ISSUE #185: [Request] Sidecar report generator
### Describe the feature

Related to #115

### Anything other information?

_No response_

### PR DESCRIPTION
**Pull request by andrewazores**
Fixes #185

~~Currently in draft mode because there is some connection flakiness ("Connection refused") on report generation requests. It seems like some misconfiguration or maybe load balancer naming conflict.~~

### CODE REVIEW

**Comment by andrewazores** [comment]
The connection flakiness actually reminds me a lot of what I observed before:

https://github.com/cryostatio/cryostat/pull/607

So it seems the underlying HTTP client library was not really the problem, or the solution.

**Comment by andrewazores** [comment]
Opened #194 to track the connection flakiness as a separate issue.

File: charts/cryostat/values.yaml [hunk 1]
```diff
@@ -82,6 +82,38 @@ core:
...
+    ## @param reports.service.httpPort Port number to expose on the Service for the Report Generator Deployment
+    httpPort: 10001
+  ## @param reports.replicas Number of Report Generator replicas to deploy. If zero, the Deployment and Service will not be created and the main Cryostat container will handle all report generations on its own.
+  replicas: 0
+  resources:
+    requests:
+      ## @param reports.resources.requests.cpu CPU resource request for each Pod in the Report Generator Deployment.
+      cpu: 2000m
+      ## @param reports.resources.requests.memory Memory resource request for each Pod in the Report Generator Deployment.
+      memory: 512Mi
```
Comment (tthvo) [inline charts/cryostat/values.yaml:107]: Should these match the specifications here?

https://github.com/cryostatio/cryostat-operator/blob/main/docs/config.md#reports-options

File: charts/cryostat/templates/cryostat_deployment.yaml [hunk 2]
```diff
@@ -57,6 +57,10 @@ spec:
             value: none
           - name: QUARKUS_HIBERNATE_ORM_SQL_LOAD_SCRIPT
             value: no-file
+          {{- if gt (int .Values.reports.replicas) 0 }}
+          - name: QUARKUS_REST_CLIENT_REPORTS_URL
+            value: {{ printf "http://%s-reports.%s.svc:%d" $fullName $.Release.Namespace (int .Values.reports.service.httpPort) }}
```
[the remark this answers is not in the document]
Comment (andrewazores) [reply charts/cryostat/templates/cryostat_deployment.yaml:62]: True, it should be unnecessary to specify the namespace and `.svc`. I think I did this out of desperation while trying to figure out what turned out to be the label selectors bug :-)

**Review by tthvo — APPROVED** [review]
Looks good to  me! There is definitely flankiness in connection between components. But when I fixed the selector as in #197, all are good :D

**Pull request merged**